src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -375,26 +375,56 @@ def find_referenced_image_set(kwargs: dict):
     return referenced_images
 
 
-def normalize_connection_config(connection):
-    """
-    Normalizes the configuration of a given connection object for compatibility.
+def init_openai_client(connection: OpenAIConnection):
+    try:
+        from openai import OpenAI as OpenAIClient
+    except Exception:
+        raise Exception(
+            "Please upgrade your OpenAI package to version 1.0.0 or later" +
+            "using the command: pip install --upgrade openai.")
+
+    return OpenAIClient(
+            # disable OpenAI's built-in retry mechanism by using our own retry
+            # for better debuggability and real-time status updates.
+            max_retries=0,
+            api_key=connection.api_key,
+            organization=connection.organization,
+            base_url=connection.base_url
+        )
 
-    This function takes a connection object and normalizes its configuration,
-    ensuring it is compatible and standardized for use.
-    """
-    if isinstance(connection, AzureOpenAIConnection):
-        return {
-            "api_key": connection.api_key,
-            "api_version": connection.api_version,
-            "azure_endpoint": connection.api_base
-        }
-    elif isinstance(connection, OpenAIConnection):
-        return {
-            "api_key": connection.api_key,
-            "organization": connection.organization,
-            "base_url": connection.base_url
-        }
+
+def init_azure_openai_client(connection: AzureOpenAIConnection):
+    try:
```

Comment:
please kindly help add the more precise exception capture change:
```
try:
    from openai import AzureOpenAI as AzureOpenAIClient
except ImportError as e:
    if "cannot import name 'AzureOpenAI' from 'openai'" in str(e):
        raise ImportError(
            "Please upgrade your OpenAI package to version 1.0.0 or later" +
            "using the command: pip install --upgrade openai.")
    else:
        raise e
```

Reply:
This could make such error can be correctly thrown.
```
ImportError: cannot import name 'OpenAI' from partially initialized module 'openai' (most likely due to a circular import) (C:\Users\chjinche\projects\promptflow\src\promptflow-tools\promptflow\tools\openai.py)
```